### ISSUE TITLE
bug(xlsx): partition_xlsx() drops text in certain edge-cases

### ISSUE THREAD

**Issue by scanny** [Unstructured-IO/unstructured#2485]
**Description**
In XLSX, we consider a contiguous block of cells separated by a blank line to be a "sub-table" which is partitioned as a `Table` element. When a subtable has a certain form of non-uniformity (still working out particulars), it is only partially emitted with the effect that much of the table text does not appear in the output.

**To Reproduce**
1. Download this file at a convenient local path: [tLists2_sum_adjusted.xlsx](https://github.com/Unstructured-IO/unstructured/files/14117693/tLists2_sum_adjusted.xlsx)
2. Run this code:
    ```python
    from unstructured.partition.xlsx import partition_xlsx
    elements = partition_xlsx("tLists2_sum_adjusted.xlsx")
    for e in elements:
        print(f"{repr(e)}: text={repr(e.text)}")
    ```
3. Note that the text of the main table is missing (should appear immediately above second-to-last element with text "Total".

**Additional context**
This appears to be related to a subtable with a "Tee" shape like this (N = "number"):
```
N  | N   | N
N  | N   | N
   | str |
```

It does not occur when the last row extends to the right extent:
```
N  | N   | N
N  | N   | N
   | str | N
```

The "Total" value cell (J21: 69,364.70) is not being picked up by `pandas`, probably because it is a formula without a cached "last-computed-value", which is what is producing the "T" shape in the original file. If the file is opened and saved without making any changes then the problem disappears.

**Comment by scanny** [comment]
Further investigation reveals that there are multiple cases where text is dropped:

```markdown
    # -- ❌ CASE 3: There are two-or-more leading single-cell rows.
    #       -> leading single-cell rows are included in subtable
    #
    #    a    -> [ Table(a, b, c, d, e, f) ]
    #    b
    #    c d
    #    e f

    # -- ❌ CASE 4: There is exactly one trailing single-cell row.
    #      -> core table is dropped. trailing single-cell row is emitted as Title
    #         (this is the behavior in the reported bug)
    #
    #    a b  -> [ Title(e) ]
    #    c d
    #      e

    # -- ❌ CASE 5: There are two-or-more trailing single-cell rows.
    #      -> core table is dropped. trailing single-cell rows are each emitted as a Title
    #
    #    a b  -> [ Title(e),
    #    c d       Title(f) ]
    #    e
    #    f

    # -- ❌ CASE 9: Single-row subtable, no single-cell rows above or below.
    #      -> First cell is mistakenly emitted as title, remaining cells are dropped.
    #
    #    a b c  -> [ Title(a) ]

    # -- ❌ CASE 10: Single-row subtable with one leading single-cell row.
    #      -> Leading single-row cell is correctly identified as title, core-table is mis-identified
    #         as a `Title` and truncated.
    #
    #    a      -> [ Title(a),
    #    b c d       Title(b) ]

```

**Comment by scanny** [comment]
Fixed by #2534
